test/test_torch.py

```diff
@@ -17596,6 +17596,29 @@ def test_multinomial_empty(self, device):
             out = torch.multinomial(probs, num_samples=num_samples, replacement=replacement)
             self.assertEqual(out, expected)
 
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @dtypes(*(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes(include_bfloat16=False)))
+    def test_count_nonzero(self, device, dtype):
+        for ndims in (0, 2, 1, 3):
+            shape = self._rand_shape(ndims, 5, 10)
+            for n in range(ndims + 1):
+                for c in combinations(list(range(ndims)), n):
+                    for count_dim in permutations(c):
+                        if shape == ():
+                            x = torch.tensor(())
+                        else:
+                            x = torch.randn(*shape)
```

Comment:
You also need to construct this with the correct device and dtype. In addition, if the dtype is not a floating type you should use randint:

```
if not dtype.is_floating_point:
  t = torch.randint(1, 2, shape, device=device, dtype=dtype)
else:
  t = torch.randn(*shape, device=device).to(dtype)  
```

In the latter case I believe you'll need to .to the result of randn because I don't think we support randn for all float types on the CPU or CUDA.

Reply:
Btw, I took a note to simplify self.compare_with_numpy() so it's clearer how to use it. Sorry about that.